internal/service/flexcluster/model.go

```diff
@@ -41,6 +41,25 @@ func NewTFModel(ctx context.Context, apiResp *admin.FlexClusterDescription202411
 	}, nil
 }
 
+func NewTFModelDSP(ctx context.Context, projectID string, input *admin.PaginatedFlexClusters20241113) (*TFModelDSP, diag.Diagnostics) {
+	diags := &diag.Diagnostics{}
+	results := input.GetResults()
+	tfModels := make([]TFModel, len(results))
+	for i := range results {
+		item := &results[i]
+		tfModel, diagsLocal := NewTFModel(ctx, item)
+		diags.Append(diagsLocal...)
+		tfModels[i] = *tfModel
```

Comment:
would add an if to ensure tfModel is defined before appending to results, this avoids a potential  nil pointer dereference

Reply:
Added